### DISCUSSION TITLE
Custom vertex attributes

### DISCUSSION THREAD

**Discussion by kamkash** [google/filament#5523] [Q&A] [answered]
The following vertex shader fails to compile.
I am trying to use the getCustom0() API to retrieve custom vertex attributes. Am I missing something?

material {
    name : BakedTexture,
    variables: [
        samplerNumber
    ],
}

vertex {
    void materialVertex(inout MaterialVertexInputs material) {
       float4 p = getCustom0();
       material.samplerNumber = p;
    }
}

matc -a all -o multiTextureUnlitTransparent.filamat multiTextureUnlitTransparent.mat

ERROR: Unable to parse vertex shader
ERROR: multiTextureUnlitTransparent.mat:26: 'getCustom0' : no matching overloaded function found 
ERROR: multiTextureUnlitTransparent.mat:26: '=' :  cannot convert from ' const float' to ' temp highp 4-component vector of float'
ERROR: multiTextureUnlitTransparent.mat:26: '' : compilation terminated 
ERROR: 3 compilation errors.  No code generated.

**Comment by romainguy** [comment, marked as answer]
You must declare the your material requires that attribute: `requires: [ custom0 ]`

**Reply by kamkash** [reply]
Thanks for the quick response!!!
